gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -191,26 +138,42 @@ static GoogleCloudStorageGrpcReadChannel open(
       GoogleCloudStorageReadOptions readOptions,
       BackOffFactory backOffFactory)
       throws IOException {
-    // The gRPC API's GetObjectMedia call does not provide a generation number, so to ensure
-    // consistent reads, we need to begin by checking the current generation number with a
-    // separate
-    // call.
-    try {
-      return ResilientOperation.retry(
-          () ->
-              openChannel(
-                  stubProvider,
-                  storage,
-                  errorExtractor,
-                  resourceId,
-                  watchdog,
-                  readOptions,
-                  backOffFactory),
-          backOffFactory.newBackOff(),
-          RetryDeterminer.ALL_ERRORS,
-          IOException.class);
-    } catch (Exception e) {
-      throw new IOException(String.format("Error reading '%s'", resourceId), e);
+    checkArgument(storage != null, "GCS json client cannot be null");
+    this.useZeroCopyMarshaller =
+        ZeroCopyReadinessChecker.isReady() && readOptions.isGrpcReadZeroCopyEnabled();
+    this.stub = stubProvider.newBlockingStub();
+    GoogleCloudStorageItemInfo itemInfo =
+        getObjectMetadata(resourceId, errorExtractor, backOffFactory, storage);
```

Comment:
We should skip this, if fail fast option is not enabled

Reply:
Until we can use footer prefetch call for metadata, We would end up with an additional metadata call. fail fast would not provide an efficient read in this case. 

This is beyond the scope of this PR. will discuss & track this offline